inc/server/class-prompt-server.php

```diff
@@ -89,15 +110,85 @@ public function register_routes() {
 		);
 	}
 
+	/**
+	 * Save the API key.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 * @return \WP_Error|\WP_HTTP_Response|\WP_REST_Response
+	 */
+	public function save_api_key( $request ) {
+		$body = $request->get_body();
+		$body = json_decode( $body, true );
+
+		if ( ! isset( $body['api_key'] ) ) {
+			return new \WP_Error( 'rest_invalid_json', __( 'API key is missing.', 'otter-blocks' ), array( 'status' => 400 ) );
+		}
+
+		$api_key = sanitize_text_field( $body['api_key'] );
+
+		if ( empty( $api_key ) ) {
+			update_option( 'themeisle_open_ai_api_key', $api_key );
+			return new \WP_REST_Response( array( 'message' => __( 'API key saved.', 'otter-blocks' ) ), 200 );
+		}
+
+		$response = wp_remote_post(
+			self::$base_url,
+			array(
+				'method'  => 'POST',
+				'headers' => array(
+					'Authorization' => 'Bearer ' . $api_key,
+					'Content-Type'  => 'application/json',
+				),
+				'body'    => wp_json_encode(
+					array(
+						'model'    => 'gpt-3.5-turbo',
+						'messages' => array(
+							array(
+								'role'    => 'system',
+								'content' => 'You are a helpful assistant.',
+							),
+							array(
+								'role'    => 'user',
+								'content' => 'Hello!',
+							),
+						),
+					)
+				),
+				'timeout' => 2 * MINUTE_IN_SECONDS,
+			)
+		);
+
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$body = wp_remote_retrieve_body( $response );
+		$body = json_decode( $body );
+
+		if ( isset( $body->error ) ) {
+			if ( isset( $body->error->message ) ) {
+				return new \WP_Error( isset( $body->error->code ) ? $body->error->code : 'unknown_error', $body->error->message );
+			}
+
+			return new \WP_Error( 'unknown_error', __( 'An error occurred while processing the request.', 'otter-blocks' ) );
```

Comment:
You can use a ternary operator here:

```php
return isset( $body->error->message ) ? new \WP_Error( isset( $body->error->code ) ? $body->error->code : 'unknown_error', $body->error->message ) : new \WP_Error( 'unknown_error', __( 'An error occurred while processing the request.', 'otter-blocks' ) );
``` 

You can do multiline if you prefer it to be more readable:

```php
return isset( $body->error->message ) 
    ? new \WP_Error( isset( $body->error->code ) ? $body->error->code : 'unknown_error', $body->error->message ) 
    : new \WP_Error( 'unknown_error', __( 'An error occurred while processing the request.', 'otter-blocks' ) );
```